android/modules/media/src/java/ti/modules/titanium/media/TiCameraXActivity.java

```diff
@@ -0,0 +1,570 @@
+				if (error != VideoRecordEvent.Finalize.ERROR_NONE) {
+					// error
+				}
+				Uri uri = finalizeEvent.getOutputResults().getOutputUri();
+
+				if (uri.toString().startsWith("content://")) {
+					String path = "";
+					String name = "";
+
+					Uri mediaUri = MediaStore.Video.Media.EXTERNAL_CONTENT_URI;
+					String[] columns = {
+						MediaStore.Video.VideoColumns.TITLE,
+						MediaStore.Video.VideoColumns.DATA
+					};
+					String[] selectionArgs = { mediaTitle };
+					String queryString = MediaStore.Video.VideoColumns.TITLE + " = ? ";
+					Cursor cursor = contentResolver.query(mediaUri, columns, queryString, selectionArgs, null);
+					if ((cursor != null) && cursor.moveToNext()) {
+						name = getStringFrom(cursor, 0);
+						path = getStringFrom(cursor, 1);
+					}
+
+					if (path != null && !path.equals("")) {
+						TiBlob blob = TiBlob.blobFromFile(TiFileFactory.createTitaniumFile(path, false));
+						KrollDict response = MediaModule.createDictForImage(blob, blob.getMimeType());
+						successCallback.callAsync(callbackContext, response);
+					}
+				} else {
+					// normal file
+					File file = new File(uri.getPath());
+					TiBlob blob = TiBlob.blobFromFile(TiFileFactory.createTitaniumFile(file.getPath(), false));
+					KrollDict response = MediaModule.createDictForImage(blob, blob.getMimeType());
+					successCallback.callAsync(callbackContext, response);
+				}
+
+				if (cameraActivity != null && autohide) {
+					if (recording != null) {
+						recording.close();
+						recording = null;
+					}
+					cameraActivity.finish();
+				}
+			}
+			// All events, including VideoRecordEvent.Status, contain RecordingStats.
+			// This can be used to update the UI or track the recording duration.
+			RecordingStats recordingStats = videoRecordEvent.getRecordingStats();
+			long recordingTime = (recordingStats.getRecordedDurationNanos() / 1000 / 1000);
+			if (isRecording && videoMaximumDuration > 0 && recordingTime >= videoMaximumDuration) {
+				stopVideoCapture();
+			}
+			if (isRecording && recordingCallback != null) {
+				recordingDict.put("duration", recordingTime);
+				recordingDict.put("size", recordingStats.getNumBytesRecorded());
+				recordingCallback.callAsync(callbackContext, recordingDict);
+			}
+		};
+
+		if (cameraActivity.hasAudioRecorderPermissions()) {
+			recording = pendingRecording.withAudioEnabled().start(executor, vre);
+			isRecording = true;
+		} else {
+			recording = pendingRecording.start(executor, vre);
+			isRecording = true;
+		}
+	}
+
+	public static void pauseVideoCapture()
+	{
+		if (recording != null) recording.pause();
+		isRecording = false;
+	}
+
+	public static void resumeVideoCapture()
+	{
+		if (recording != null) recording.resume();
+		isRecording = true;
+	}
+
+	public static void stopVideoCapture()
+	{
+		if (recording != null) recording.stop();
+		isRecording = false;
+	}
+
+	public static void setFlashMode(int cameraFlashMode)
+	{
+	}
```

Comment:
Add implementation or remove / add note?

Reply:
Missed that, will implement it!